src/main/java/net/neoforged/neoforge/event/entity/player/ItemTooltipEvent.java

```diff
@@ -12,9 +12,10 @@
 import net.minecraft.world.item.Item.TooltipContext;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.TooltipFlag;
+import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.Nullable;
 
-public class ItemTooltipEvent extends PlayerEvent {
+public abstract class ItemTooltipEvent extends PlayerEvent {
```

Comment:
I'd suggest keeping this as a single class with an enum parameter. Look at RenderLevelStageEvent for an example.

Reply:
If this is made into an enum, all existing mods will suddenly duplicate the added info multiple times, and it'll look like the mods' fault.

Reply:
Futher discussed over in [discord](https://discord.com/channels/313125603924639766/1154167065519861831/1244559633771663383) in which multiple events won out the discussion